lib/crypto/src/crypto.erl

```diff
@@ -1420,11 +1420,21 @@ verify(Algorithm0, Type0, Data, Signature, Key, Options) ->
 pkey_verify_nif(_Algorithm, _Type, _Data, _Signature, _Key, _Options) -> ?nif_stub.
 
 %% Backwards compatible:
-sign_verify_compatibility(dss, none, Digest) ->
+sign_compatibility(dss, none, Digest) ->
     {sha, {digest, Digest}};
-sign_verify_compatibility(Algorithm0, Type0, _Digest) ->
+sign_compatibility(Algorithm0, Type0, _Digest) ->
     {Algorithm0, Type0}.
 
+internal_verify_alg_types(dss, none, Digest) ->
+    %% Backwards compatible
+    {sha, {digest, Digest}};
+internal_verify_alg_types(eddsa, none, _) ->
+    {eddsa, undefined};
```

Comment:
The change from 'none' to 'undefined' should not be needed afaik, because if the sign/verify algorithm name is 'eddsa' the hash type argument ( none,undefined ) not checked.

Reply:
I also think that it is not necessary with two different functions for the same purpose, that is, adapting the sha and digest for old programs to the new format. And it is the same C-function that handles those parameter in the nif (pkey.c: get_pkey_sign_digest)

Reply:
Humm ... I tried removing it altogether and it seems to work as you suggest @HansN . I am sure I needed it at some point during the development but it might have been intermediate. There was many steps and rebases before this started to feel like the solution.